### PR TITLE
🐛Source Amplitude: fix start date validation

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
https://github.com/airbytehq/oncall/issues/438 - Source: Amplitude - requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://amplitude.com/api/2/users?m=active&i=1&g=country&s... (via Sentry)

Field start date set in future.

## How
Add validation to stream method and if start date in future start date set to current date and inform customer in logs about that. 

## Pre-merge Checklist
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by arsenlosenko — APPROVED** [review]

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/source.py [hunk 1]
```diff
@@ -29,10 +30,24 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
         except Exception as error:
             return False, f"Unable to connect to Amplitude API with the provided credentials - {repr(error)}"
 
+    @staticmethod
+    def _validate_start_date(start_date):
+        today = pendulum.now()
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/source.py:35]: nit: I think calling this variable `now` would be more accurate than `today`

**Review by brianjlai — APPROVED** [review]
small nits but looks good to me

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/source.py (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/source.py]: updated

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 2]
```diff
@@ -131,6 +131,7 @@ def request_params(
                     "end": self._get_end_date(start_datetime).strftime(self.date_template),
                 }
             )
+        print(params)
```
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py]: updated

**Pull request merged**